File: prob10.c
#include <stdio.h>

#define odd(x) ((x & 1) ? (x) : (x = (x + 1)))

unsigned long int prime_gen(int start){

	unsigned int k, found = 1;
	
	if(start == 2)
		return 2;

	for(odd(start); ; start += 2, found = 1){		// all even numbers are not prime
		for(k = 2; k < start; ++k)
			if(!(start % k))
				found = 0;
		if(found)
			break;
	}

	return start;
}

int main(){

	long double tmp = 0, aux = 2;

	for(; aux < 1000000; aux = prime_gen(++aux)){
		tmp += aux;
	}

	printf("Result : %Lf\n", tmp);

	return 0;

}


/*

// quick version

#include <stdio.h>
#define TRUE 1
#define FALSE 0
#define MILLION 2000000
 
int main( int argc, char* argv[] ) {
	long long unsigned sum;
	int nums[MILLION], i, j;
	sum = 0;
 
	for( i = 0; i < MILLION; i++ )
		nums[i] = 1;
 
	printf("Array complete.\n");
 
	// now we transform the multiples of n into 0, so they don't interfere
	// in the sum:
 
	for( i = 2; i < MILLION; i++ ) {
		if(nums[i]) {
			for( j = i*2; j < MILLION; j += i) {
				nums[j] = 0;
			}
		}	
	}
 
	printf("Removal of compose numbers complete.\n");
 
	for( i = 2; i < MILLION; i++ ) {
		if(nums[i]) {
			sum += i;
			//printf("Summing: %i\n",i);
		}
	}
 
	printf("%llu\n",sum);
 
	return 0;
}

*/

File: prob10.java

class prob10 {

	static long odd(long x){
		if((x % 2) == 0)
			return (x + 1);
		else
			return x;
	}

	static long prime_gen(long start){

		int k;
		boolean found = true;
	
		if(start == 2)
			return 2;

		for(odd(start); ; start += 2, found = true){		// all even numbers are not prime
			for(k = 2; k < start; ++k)
				if((start % k) == 0)
					found = false;
			if(found)
				break;
		}

		return start;
	}

	public static void main(String[] args){
		long tmp = 0l, aux = 2l;

		for(; aux < 11; aux = prime_gen(++aux)){
			tmp += aux;
			System.out.println("aux : " + aux + ", tmp : " + tmp + "\n");
		}

		System.out.println("Result : " + tmp + "\n");

	}

}


